tests/backup/backup_kubevirt_test.go

```diff
@@ -2105,3 +2106,280 @@ var _ = Describe("{KubevirtVMRestoreWithAfterChangingVMConfig}", Label(TestCaseL
+					log.InfoD("Scheduled VM [%s] in source cluster in namespace [%s]", appName, namespace)
+					scheduledAppContexts = append(scheduledAppContexts, appCtx)
+				}
+			}
+		})
+
+		Step("Validating applications", func() {
+			// To Do: Add Data validation for this test case
+			log.InfoD("Validating applications")
+			ValidateApplications(scheduledAppContexts)
+		})
+		wg.Add(2)
+		go func() {
+			defer GinkgoRecover()
+			defer wg.Done()
+			Step("Verify the recreated VM is included in next non namespace labelled scheduled backup", func() {
+				log.InfoD("Verify the recreated VM is included in next non namespace labelled scheduled backup")
+				for _, namespace := range bkpNamespaces {
+					schBackupAfterVMcreation, err := GetNextScheduleBackupName(nonLabelSchNamespaceMap[namespace], time.Duration(periodicSchedulePolicyInterval), ctx)
+					dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying fetching next schedule backup [%s] after VM creation ", schBackupAfterVMcreation))
+					appContextsToBackup := FilterAppContextsByNamespace(scheduledAppContexts, []string{namespace})
+					err = GetUpdatedKubeVirtVMSpecForBackup(appContextsToBackup)
+					dash.VerifyFatal(err, nil, fmt.Sprintf("Verification of getting the updated kubevirt VM context"))
+					err = BackupSuccessCheckWithValidation(ctx, schBackupAfterVMcreation, appContextsToBackup, BackupOrgID, MaxWaitPeriodForBackupCompletionInMinutes*time.Minute, 30*time.Second)
+					dash.VerifyFatal(err, nil, fmt.Sprintf("Verification of success of next VM schedule backup [%s] of schedule %s", schBackupAfterVMcreation, nonLabelSchNamespaceMap[namespace]))
+				}
+			})
+		}()
+		go func() {
+			defer GinkgoRecover()
+			defer wg.Done()
+			Step("Verify the recreated VM is included in next namespace labelled scheduled backup ", func() {
+				log.InfoD("Verify the recreated VM is included in next namespace labelled scheduled backup ")
+				for _, namespace := range bkpNamespaces {
+					labelledSchBackupAfterVMcreation, err := GetNextScheduleBackupName(labelSchNamespaceMap[namespace], time.Duration(periodicSchedulePolicyInterval), ctx)
+					dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying fetching next labelled schedule [%s] backup after VM creation ", labelledSchBackupAfterVMcreation))
+					appContextsToBackup := FilterAppContextsByNamespace(scheduledAppContexts, []string{namespace})
+					err = GetUpdatedKubeVirtVMSpecForBackup(appContextsToBackup)
+					dash.VerifyFatal(err, nil, fmt.Sprintf("Verification of getting the updated kubevirt VM context"))
+					err = BackupSuccessCheckWithValidation(ctx, labelledSchBackupAfterVMcreation, appContextsToBackup, BackupOrgID, MaxWaitPeriodForBackupCompletionInMinutes*time.Minute, 30*time.Second)
+					dash.VerifyFatal(err, nil, fmt.Sprintf("Verification of success of next VM labelled schedule backup [%s] of schedule %s", labelledSchBackupAfterVMcreation, labelScheduleName))
+				}
+			})
+		}()
+		wg.Wait()
+	})
```

Comment:
Can you have single step which runs these 2 go routines instead of running the steps itself in a go routine?
I am not sure if Step is thread safe